core/ledger/kvledger/pvtrwstorage/transient_store.go

```diff
@@ -16,26 +16,154 @@ limitations under the License.
+	return &transientStoreProvider{dbProvider: dbProvider}
+}
+
+// OpenStore returns a handle to a name database in TransientStore
+func (provider *transientStoreProvider) OpenStore(dbName string) (TransientStore, error) {
+	dbHandle := provider.dbProvider.GetDBHandle(dbName)
+	return &transientStore{db: dbHandle, dbName: dbName}, nil
+}
+
+// Close closes the TransientStoreProvider
+func (provider *transientStoreProvider) Close() {
+	provider.dbProvider.Close()
+}
+
+// Persist stores the private read-write set of a transaction in the transientStore
+func (store *transientStore) Persist(txId string, endorserId string,
+	endorsementBlkHt uint64, privateSimulationResults []byte) error {
+
+	// Create compositeKey with the namespace of private read-write set, txId and endorserId
+	compositeKey := CreateCompositeKeyForPRWSet(PRWSET_NS, txId, endorserId, endorsementBlkHt)
+	if err := store.db.Put(compositeKey, privateSimulationResults, true); err != nil {
+		return err
+	}
+
+	// Create compositeKey with endorsementBlkHt, txId, endorserId & Store the compositeKey (purge index)
+	// in a separate namespace with a null byte as value.
+	compositeKey = CreateCompositeKeyForPurgeIndex(PURGEINDEX_NS, endorsementBlkHt, txId, endorserId)
+	if err := store.db.Put(compositeKey, emptyValue, true); err != nil {
+		return err
+	}
+	return nil
+	// TODO: Node failure can result in inconsistent state between private read-write set namespace
+	// and purge index. Need to recover from failure.
+}
+
+// GetTxPrivateRWSetbyTxId returns private read-write set of a given transaction
+func (store *transientStore) GetTxPrivateRWSetByTxId(txId string) (commonledger.ResultsIterator, error) {
+	startKey := CreatePartialCompositeKeyForPRWSet(PRWSET_NS, txId, false)
+	endKey := CreatePartialCompositeKeyForPRWSet(PRWSET_NS, txId, true)
+	iter := store.db.GetIterator(startKey, endKey)
+	return &rwsetScanner{txId, iter}, nil
+}
+
+// Purge removes private read-writes set generated by endorsers at block height lesser than
+// a given maxBlockNumToRetain. In other words, Purge only retains private read-write sets
+// that were generated at block height of maxBlockNumToRetain or higher.
+func (store *transientStore) Purge(maxBlockNumToRetain uint64) error {
+	// Do a range query with 0 as startKey and maxBlockNumToRetain
+	startKey := CreatePartialCompositeKeyForPurgeIndex(PURGEINDEX_NS, 0, false)
+	endKey := CreatePartialCompositeKeyForPurgeIndex(PURGEINDEX_NS, maxBlockNumToRetain, true)
+	iter := store.db.GetIterator(startKey, endKey)
+
+	// get all txId and endorserId from above result and remove it from the db
+	for iter.Next() {
+		dbKey := iter.Key()
+		txId, endorserId, endorsementBlkHt := SplitCompositeKeyOfPurgeIndex(dbKey)
+		compositeKey := CreateCompositeKeyForPRWSet(PRWSET_NS, txId, endorserId, endorsementBlkHt)
+		if err := store.db.Delete(compositeKey, true); err != nil {
+			return err
+		}
+		if err := store.db.Delete(dbKey, true); err != nil {
+			return err
+		}
```

Comment:
Again use a batch and keep adding deletes in the batch within this for loop. Write the batch at the end of for loop